### DISCUSSION TITLE
Lightweight IDE, to make development much better and comfortable

### DISCUSSION THREAD

**Discussion by ddovnar** [apache/netbeans#7666] [Ideas]
nowadays, we have powerful IDEA (such as Intellij IDEA, or last version of Netbeans), but they have a general problem - they are heavy to load and require many resources (CPU, RAM). Sometimes developers need a good IDE (most likely to editor), in which they can to do mostly work: to open project, and do editing without code intellijents. I'm trying to use Intellij IDEA in the laptop working on battery, and I can't to this...the battery fast discharging and CPU in loading, the cooler was working always.

Idea: Is it possible to create lightweight IDE based on Netbeans platform? The idea is to use old beautiful version 8.2 with support JDK8, but only is need to do for that - is to make JDK17+ to support. How can we do this?

**Comment by ddovnar** [comment]
I have done this experiment, and created a build Netbeans 8.2 with support modern JDK 22 features. It's work very good for me and my laptop battery not used unnecessary. At the end, I have a good lightweight IDE, really cool. There are some screenshots for demo:

![netbeans82_jdk22](https://github.com/user-attachments/assets/2aa95a99-0ad6-49a2-92a9-53630b006143)

![netbeans_resources](https://github.com/user-attachments/assets/44c4c274-0063-487b-bb13-6ac3c5172f14)

**Comment by ddovnar** [comment]
Comparing to Intellij IDEA, I just launch IDEA and open some project, the situation with resources is not good:
![intellij_idea_resources](https://github.com/user-attachments/assets/0c5612fc-0ebf-45ad-9d71-cf992d067e5c)

**Comment by wumpz** [comment]
Just out of couriosity, how do resources compare between Netbeans 8.2 and Netbeans 22?
